pkg/network/protocols/kafka/stats.go

```diff
@@ -66,24 +91,72 @@ func (r *RequestStats) CombineWith(newStats *RequestStats) {
 			// Nothing to do in this case
 			continue
 		}
-		r.AddRequest(statusCode, newRequests.Count, newRequests.StaticTags)
+
+		if newRequests.Latencies == nil {
+			// In this case, newRequests must have only FirstLatencySample, so use it when adding the request
+			r.AddRequest(statusCode, newRequests.Count, newRequests.StaticTags, newRequests.FirstLatencySample)
+			continue
+		}
+
+		stats, exists := r.ErrorCodeToStat[statusCode]
+		if !exists {
+			stats = &RequestStat{}
+			r.ErrorCodeToStat[statusCode] = stats
+		}
+		// The other bucket (newStats) has a DDSketch object
+		// We first ensure that the bucket we're merging to have a DDSketch object
+		if stats.Latencies == nil {
+			stats.Latencies = newRequests.Latencies.Copy()
+
+			// If we have a latency sample in this bucket we now add it to the DDSketch
+			if stats.FirstLatencySample != 0 {
+				err := stats.Latencies.AddWithCount(stats.FirstLatencySample, float64(stats.Count))
+				if err != nil {
+					log.Debugf("could not add kafka request latency to ddsketch: %v", err)
```

Comment:
Should these errors be propagate to the caller and/or have a higher log level?

Reply:
https://datadoghq.atlassian.net/browse/USMON-1142